### DISCUSSION TITLE
Is possible do a GPU Undervolt on AMD radeon 6800m of 2021 g15?

### DISCUSSION THREAD

**Discussion by sanslash332** [seerge/g-helper#3250] [Q&A] [answered]
Hello guys,
A small questions about the undervolting feature of the Ghelper. I saw on the FAQ the explanation about the undervolting of the GPU, but... I have some confusion because the UI and well if my GPU support that...

Let me explain some things first.

1. I'm a totally blind user. So, for control the PC, I depend of a assistive technology tool, specifically a screen reader. I use JAWS for windows as my main screen reader, and NVDA (non visual desktop access) as a support screen reader, when the first crash or found a application that doesn't work well with it.
2. My laptop is a 2021 strix g15 advantage edition, with the r9 5900hx processor, radeon 6800m GPU and gHelper v0.195 installed.

So, with that, small questions:

1. Is undervolteable with gHelper my GPU?
2. where find the sliders of "core clock offset" and "core clock limmit" that are mentioned on the FAQ?

Under the section "fans and power"
navigating with tab, I just found these things:

1. aparently two charts, that says "chart CPU" and "chart GPU". They aren't interactuables with keyboard
2. A combo box with the built-in presets of fan curbes
3. Next to the combo box, is a button that doesn't habe any label or indicator readable with the screen reader.
4. After that a button to restore factory values.
5. next the checkbox that says "apply personaliced curbe"
6. next the calibrate button
7. after that a combo box that doesn't habe a explanatory label. Just the posible values are disabled, enabled, agresive, eficient enabled, eficient aggresive, and eficient and aggresive at guaranteed... ¿what are that combo box) ¿what it changes?
8. Next are two sliders withouth labels. One aparently is a percentage of something that I have set on 30, And I guess that sais "platform (CPU + GPU) the second is aparently the limmit power of the CPU I set it at 78 the slider value, and the name changes to 80w
9. Next a checkbox that says "Apply power limmits"
10. A combo box that says "CPU boost"
11. And finally two buttons that says CPU and advanced.
If you press CPU it does nothing, but when you press advanced, appear two new sliders. One that says undervolt and are for CPU (aparently), the temp limmit slider, and the apply button. All of these are for CPU. And with advanced open, if you press that CPU button, returns to the previows window.

So I've missing something? or my GPU isn't undervolteable under gHelper?

Also, please take this question a small like accessibility review.

Some things require better labeling or exposition to the accessibility apis of microsoft, but in general the app is very usable with screen readers.

Thanks for your amazing job!

**Comment by seerge** [comment, marked as answer]
@sanslash332 hello,

That's nice that NVDA works with the app, I was explicitly adding narrator support to the app by requests from other blind user. Check this issues https://github.com/seerge/g-helper/issues/1823 https://github.com/seerge/g-helper/issues/1830

To answer your main questions :

1. AMD GPUs are totally locked and can't be undervolted or overclocked. That's how things are on hardware level.
2. You won't see "core clock offset" and "memory clock offset" sliders under GPU tab as they are for Nvidia GPU only.

As for your review and remarks in regard to "Fans + Power" section. 

It's indeed contains a lot of custom UI elements, that are very difficult or some times impossible to adapt to the narrators. For example CPU and GPU fan curves can be edited using chart component (with plot and dots on it). This relies on manual dragging of fan curve dots, therefore there is no realistic keyboard only alternative for this element.

And this is why whole "Fans + Power" section is not very adapted for narrators unlike main window or "Extra" settings. I simply can't adapt whole thing so I didn't see much sense in adapting only part of them. Luckily all 3 default modes (Silent, Balanced and Turbo) are absolutely fine out of the box, and work just fine in most of the cases.

Do you have any needs in setting custom power limits for your laptop? If so - I can try to add more accessible labels to the power sliders under "Fans + Power"?

**Reply by sanslash332** [reply]
Hello @seerge, again thanks for your fast answer :-)

Well... First of all. So sad for AMD GPUs... I'm finding to undervolt my graphic card, because is shuting off my laptop when its under some extrange cases. So I wanna test it if a problem of temperatures, memori overflow or something similar. But for now I cant got a satisfying conclusion.

Googling I found that exist a method to undervolt the GPU, but is a lot of hacki, tweacking the firmware using morePowerTool and well, aparently isn't easy and probably not accessible.

Talking about accessibility... nop! I'm not using NVDA to check gHelper. Specifically I used JAWS for windows.

After you mention NVDA I tested the app again, and I found some differences between NVDA and JAWS.

The most important, is that with NVDA the sliders under fan and power section, aren't directly readable.
You can focus it (without knowing that you did) and move them with the arrow keys, but after change them you've to review them using the special "object navigator" of NVDA that isn't the default method to check controls on the app.

So, Please try to check what apen with the exposition of these sliders, why works fine with JAWS, but not with NVDA.

And... what is that combo box that have options like eficient, aggresive, etc?

Please add a label to it to know what setting are you changing with it.

And yes please. If you can tune the most parts of that window, do it.
I know the issues with the flow charts, but other things like sliders, buttons, combo boxes and whatever try to keep them the more accessible as you can. Is fun (and very util) browse through these options to customize your machine.


About the chart itself. Hum. Is a interesting challenge how accessibilice that kind of information. Maybe is doable with sliders or directly edit boxes with values of the coordinates for each of the points on the chart, but obviously you need a lot of maths knowledge if you want to tune your pc using that kind of information 🤣 

Again. thanks for all.

Very amazing app.

**Reply by sanslash332** [reply]
@seerge 

Small question. under fans and power window what is the difference between the two sliders that appear on that window to control the power?

One says CPU, and other platform CPU+GPU.

What do these sliders and how work each one?

Thanks

**Reply by seerge** [reply]
@sanslash332 hello,

- Platform (CPU + GPU) slider can define a combined power limit that your CPU and GPU can use together. With this slider you can indirectly lower maximum GPU power and make it run slower but cooler.
- CPU slider defines only CPU power limit

Those settings are typical for any all-AMD models.

You also need to check "Apply power limits" checkbox under same screen to make them work

**Reply by sanslash332** [reply]
O thanks @seerge 
It works fine.

That slider are the same TDP control that you can do in armoury crate?

**Reply by seerge** [reply]
@sanslash332 yes, those sliders are same as in Armoury Crate's Manual model. How do you use Armoury by the way? Do narrator apps work with it (cause i thought UI there is purely graphical)

**Reply by sanslash332** [reply]
Hello @seerge 

UFF, Armouri crate and screen readers.

In simple words, that app is a mess in accessibility terms.
Some basic things works, but other are a disaster.

If you wana put a score in accessibility for both applications, g-helper got a 8.5 or 9. Just are some small things that you need to tweak,  but in general terms is awesome.
In other hand, armouri crate.... got a 4-5?

Is usable for do the most basic thinks. like change your energy plan between the main 5 options (windows, manual, silent, performance and turbo), but when you want go more deeply on the app. Good luck.

Is a odyssey try change things like AI noise reduction, check GPU temps, power consumtion, fan speeds, or wathever.

Some things doesn't are identified with the correct control to iinteract with it, aren't good indicated the things that are active or inactive. Sometimes the app works like a regular program but others the UI open a sub interface that act like a webpage with the most elements on that bad tagged... Simply a headhache.
Don't talk about to manipulate manual things like aura sinc, or the manual plan. Are impossible.

A lot of things had to be used using a OCR extra tool, more fancy functions of the screen readers to try to manipulate the app. Noup, very bad.

Because that, thanks for create g-helper, and... well, take care about accessibility; im a developer too, and isn't really hard take care about these small things, and a lot of blind users glad you for these things :-)
